### PR TITLE
Create travis configuration file for Travis CI

### LINKED ISSUE #108: Create Travis CI for continuous integration
### Expected behavior 
Add Travis CI for our CI / CD environment

### Actual behavior
Currently, no CI / CD configured on this repo

### PR DESCRIPTION
**Pull request by awentzel**
For [details on formatting](https://fluentweb.com/prototyping/contribution-policy#pull-request-guidance) pull requests.

closes #108

### CODE REVIEW

**Comment by awentzel** [comment]
@nicholasrice can you come in here and add a commit to this to update the configuration to resolve the build error.  

> @microsoft/fast-dna@1.0.0 test /home/travis/build/Microsoft/fast-dna
> mocha -r ts-node/register build/**/*.spec.ts --timeout 15000
/home/travis/build/Microsoft/fast-dna/node_modules/mocha/bin/mocha:10
const spawn = require('child_process').spawn;
^^^^^
SyntaxError: Use of const in strict mode.
at Module._compile (module.js:439:25)
at Object.Module._extensions..js (module.js:474:10)
at Module.load (module.js:356:32)
at Function.Module._load (module.js:312:12)
at Function.Module.runMain (module.js:497:10)
at startup (node.js:119:16)
at node.js:945:3
npm ERR! Test failed. See above for more details.

**Comment by awentzel** [comment]
Until this issue is resolved I need to put the repo back into private status.  So when a fix is ready for this we can toggle that back to public, let Travis run to pass.  Then we can put back into private until we vet the Fluent stuff is acceptable for public audience.  After which we'll stay public.

**Comment by chrisdholt** [comment]
Let's see if installing node as ^v.4 will work here. Looks like it's installing as v0.10 which I don't believe supports ES6 syntax such as const

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+language: node_js
+node_js:
+  - "lts/*"
+dist: trusty
+sudo: false
+cache:
+  directories:
+  - node_modules
```
Comment (nicholasrice) [inline .travis.yml]: Does travis CI allow you to regex this so it caches all node_modules? Easier to maintain as we add packages.

Comment (awentzel) [reply .travis.yml]: I can look into this and add a feature task for it later.  This is how create-React-app was doing it.  But, I like your idea.

Comment (chrisdholt) [reply .travis.yml]: +1; up to you whether you want to add the task or do it now. I'd pri the task accordingly though so we can clean this up quickly.

Comment (awentzel) [reply .travis.yml]: Didn't find anything about regex.  further reading makes this sound like a bad idea with not much gain on perf if over few hundred megs.  will remove cache for now.

**Pull request merged**